tflite2onnx/op/squareddifference.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import tflite
+import numpy as np
+
+from tflite2onnx import mapping
+from tflite2onnx.op.common import Operator
+from tflite2onnx.op.binary import Binary
+
+logger = logging.getLogger('tflite2onnx')
+
+
+# wrappers are used here to override Binary.type property
+# otherwise, Binary.type fails when trying to get self.tflite
+class SubtractWrapper(Binary):
+    @property
+    def type(self):
+        return 'Sub'
+
+
+class PowerWrapper(Binary):
+    @property
+    def type(self):
+        return 'Pow'
+
+
+class SquaredDifference(Operator):
+    # use identity to propagate difference output to square input
+    TypeMapping = {
+        tflite.BuiltinOperator.SQUARED_DIFFERENCE: 'Identity',
+    }
+
+    def __init__(self, TFactory, index):
+        super().__init__(TFactory, index)
+        self.setInited()
+
+    @property
+    def type(self):
+        if self.status.uninitialized:
+            return 'Identity'
+        else:
+            op = self.tflite
+            opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+            assert(opcode in self.TypeMapping)
+            return self.TypeMapping[opcode]
+
+    def parse(self):
+        logger.debug("Parsing %s...", self.type)
+
+        op = self.tflite
+        opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+        assert(opcode in self.TypeMapping)
+
+        assert(op.InputsLength() == 2)
+        assert(op.OutputsLength() == 1)
+
+        self.parseInput(0)
+        self.parseInput(1)
+        self.parseOutput(0)
+
+        assert(len(self.inputs[0].shape) == len(self.inputs[1].shape))
+
+        # squared difference consists of two operations
+        self.prependSubtraction()   # difference
+        self.appendSquare()         # and square
+
+        self.setParsed()
+
+    def propagatableTensors(self):
+        return self.inputs + self.outputs
+
+    def transform(self):
+        pass
+
+    def prependSubtraction(self):
```

Comment:
We can remove this if we make the ONNX pattern as `Sub + Pow` only.

Reply:
https://github.com/jackwish/tflite2onnx/pull/48/files#diff-1e482a50b4377b2620d1340cb86b9dbec61b7ccb8a7e7620955a16451624d943R21-R23